enumerable.rb

```diff
@@ -0,0 +1,89 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    size.times { |n| yield (self[n]) }
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    size.times { |n| yield self[n], n }
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result_array = []
+    my_each { |n| result_array << n if yield(n) }
+    result_array
+  end
+
+  def my_all
+    return true unless block_given?
```

Comment:
### **my_all**

- [ ] It should return true if the block never returns false or nil

- [ ] It should not mutate the original array

- [ ]  When no block or argument is given it should return true when none of the collection members are false or nil

- [ ] When a class is passed as an argument returns true if all of the collection is a member of such class

- [ ] When a Regex is passed as an argument returns true if all of the collection matches the Regex

- [ ] When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

Reply:
all ruby doc tests passed, no coded changed
![test all](https://user-images.githubusercontent.com/51087315/83947958-eb676e00-a822-11ea-8dee-0bb03f47f91f.png)